code/modules/clothing/suits/wiz_robe.dm

```diff
@@ -180,4 +180,48 @@
 	src.robe_charge = FALSE
 	sleep(30)
 	src.robe_charge = TRUE
-	usr << "<span class='notice'>\The robe hums, its internal magic supply restored.</span>"
\ No newline at end of file
+	usr << "<span class='notice'>\The robe hums, its internal magic supply restored.</span>"
+
+
+//Shielded Armour
+
+/obj/item/clothing/suit/space/hardsuit/shielded/wizard
+	name = "battlemage armour"
+	desc = "Not all wizards are afraid of getting up close and personal."
+	icon_state = "battlemage"
+	item_state = "battlemage"
+	recharge_rate = 1
+	current_charges = 15
+	recharge_cooldown = INFINITY
+	shield_state = "shield-red"
+	shield_on = "shield-red"
+	min_cold_protection_temperature = ARMOR_MIN_TEMP_PROTECT
+	max_heat_protection_temperature = ARMOR_MAX_TEMP_PROTECT
+	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/shielded/wizard
+	armor = list(melee = 30, bullet = 20, laser = 20, energy = 20, bomb = 20, bio = 20, rad = 20)
+	slowdown = 0
+
+/obj/item/clothing/head/helmet/space/hardsuit/shielded/wizard
+	name = "battlemage helmet"
+	desc = "A suitably impressive helmet.."
+	icon_state = "battlemage"
+	item_state = "battlemage"
+	min_cold_protection_temperature = ARMOR_MIN_TEMP_PROTECT
+	max_heat_protection_temperature = ARMOR_MAX_TEMP_PROTECT
+	armor = list(melee = 30, bullet = 20, laser = 20, energy = 20, bomb = 20, bio = 20, rad = 20)
+
+/obj/item/wizard_armour_charge
+	name = "battlemage shield charges"
+	desc = "A powerful rune that will increase the number of hits a suit of battlemage armour can take before failing.."
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "electricity2"
+
+/obj/item/wizard_armour_charge/afterattack(obj/item/clothing/suit/space/hardsuit/shielded/wizard/W, mob/user)

```

Comment:
I always thought that `afterattack` was worse because it was called whenever you hit an atom at all instead of just hitting a certain one, that being `attackby`. Oh, well. You know best here so I'll drop the subject.